src/cmd/ksh93/sh/lex.c

```diff
@@ -256,7 +256,9 @@ int sh_lex(Lex_t* lp)
 	if(lp->noreserv)
 	{
 		lp->lex.reservok = 0;
-		while((fcgetc(c)) && c==' ' || c== '\t' || c=='\n');
+		while((fcgetc(c)) && c==' ' || c== '\t' || c=='\n')
```

Comment:
I know this is original code, but is that loop even correct? Should it not be

```c
		while(fcgetc(c) && (c==' ' || c== '\t' || c=='\n'))
```

Reply:
That loop is indeed wrong (it generated a `-Wparentheses` warning). That problem has been fixed.

Reply:
Given how fcgetc() is defined, I suppose it's actually equivalent in practical terms. Still, I think that use of parentheses is odd enough to confuse.